engine/src/main/java/org/hibernate/validator/internal/util/DomainNameUtil.java

```diff
@@ -23,36 +23,67 @@
+
 	private static final String DOMAIN = DOMAIN_LABEL + "+(\\." + DOMAIN_LABEL + "+)*";
+
 	private static final String IP_DOMAIN = "[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}";
 	//IP v6 regex taken from http://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses
 	private static final String IP_V6_DOMAIN = "(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))";
 
 	/**
-	 * Regular expression for the domain part of an email address (everything after '@')
+	 * Regular expression for the domain part of an URL
 	 */
+	// A host string must be a domain string, an IPv4 address string, or "[", followed by an IPv6 address string, followed by "]".
 	private static final Pattern DOMAIN_PATTERN = Pattern.compile(
-			DOMAIN + "|" + IP_DOMAIN + "|" + "\\[IPv6:" + IP_V6_DOMAIN + "\\]", CASE_INSENSITIVE
+			DOMAIN + "|\\[" + IP_V6_DOMAIN + "\\]", CASE_INSENSITIVE
+	);
+
+	/**
+	 * Regular expression for the domain part of an email address (everything after '@')
+	 */
+	private static final Pattern EMAIL_DOMAIN_PATTERN = Pattern.compile(
+			DOMAIN + "|\\[" + IP_DOMAIN + "\\]|" + "\\[IPv6:" + IP_V6_DOMAIN + "\\]", CASE_INSENSITIVE
 	);
 
 	private DomainNameUtil() {
 	}
 
 	/**
+	 * Checks validity of domain name used in email. To be valid it should be either valid host name, or IP address in []
+	 *
+	 * @param domain domain to check for validity
+	 *
+	 * @return {@code true} if provided string is a valid domain, {@code false} otherwise
+	 */
+	public static boolean isValidEmailDomainAddress(String domain) {
+		return isValidDomainAddress( domain, EMAIL_DOMAIN_PATTERN );
+	}
+
+	/**
 	 * Checks validity of domain name.
 	 *
 	 * @param domain domain to check for validity
 	 *
 	 * @return {@code true} if provided string is a valid domain, {@code false} otherwise
 	 */
 	public static boolean isValidDomainAddress(String domain) {
+		return isValidDomainAddress( domain, DOMAIN_PATTERN );
+	}
+
+	private static boolean isValidDomainAddress(String domain, Pattern pattern) {
 		// if we have a trailing dot the domain part we have an invalid email address.
 		// the regular expression match would take care of this, but IDN.toASCII drops the trailing '.'
 		if ( domain.endsWith( "." ) ) {
 			return false;
 		}
 
+		Matcher matcher = pattern.matcher( domain );
+		if ( !matcher.matches() ) {
+			return false;
+		}
+
```

Comment:
I'm a bit concerned about this change. To validate the domain name with the regexp, I think we should convert to punycode first. At least, that's what I see everywhere.

Or you did change the regexp to also accept the range of characters from IDN?

All in all, I think the commit would merit a longer commit message so that we can understand what you're doing.